tests/bettertransformer/testing_utils.py

```diff
@@ -175,6 +175,24 @@ def _test_raise_train(self, model_id: str, **kwargs):
             bt_model.train()
             _ = bt_model(**inputs)
 
+    def _test_train_decoder(self, model_id: str, **kwargs):
+        r"""
+        A tests that checks if the training works as expected for decoder models.
+        """
+        inputs = self.prepare_inputs_for_class(model_id=model_id, **kwargs)
+
+        hf_random_model = AutoModel.from_pretrained(model_id).eval()
+
+        bt_model = BetterTransformer.transform(hf_random_model, keep_original_model=True)
+        bt_model.train()
+        output = bt_model(**inputs)[0]
+        loss = output.sum()
+        loss.backward()
+
+        # check if gradients are not None
+        for param in bt_model.parameters():
+            self.assertIsNotNone(param.grad)
```

Comment:
Looks fine to me for now - we will still need to have a thorough convergence test at some point, but it is fine for now. See https://arxiv.org/pdf/2205.14135.pdf#figure.caption.19